### PR TITLE
Deduplicate get new identifier

### LINKED ISSUE #7016: Deduplicate get_new_identifier
Like #6984, get_new_identifier is duplicated in event, order and event invoice. It should be refactored and reused

### PR DESCRIPTION
**Pull request by jonodrew**
Fixes #7016

#### Short description of what this resolves:
This refactors a small method that's used across the software to generate an identifier

#### Changes proposed in this pull request:
In this PR I propose
- adding a generic method to the base class called `get_new_identifier`, which is basically a factory that calls the model-appropriate method to generate a new identifier
- remove repetitions of the code throughout
- change the `SocialLink` identifier to be the same as the `Event` identifier - I'm not sure this was in scope, but it seemed neater. Can be refactored if not the right call.

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The unit tests pass locally with my changes 
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/models/base.py  3
         

Complexity decreasing per file
==============================
+ app/models/social_link.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536822&bid=18150506)

File: app/models/utils.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import os
+import uuid
 import warnings
 from datetime import datetime, timedelta
 
 from sqlalchemy import Table, event, types
+from app.models import db
```
Comment (niranjan94) [inline app/models/utils.py:7]: Codacy Issue found: [Unused db imported from app.models](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536822)

**Comment by iamareebjamal** [comment]
Please solve hound and codacy issues

**Comment by iamareebjamal** [comment]
Also pytype build is failing, please check that

File: app/models/base.py [hunk 2]
```diff
@@ -9,3 +13,44 @@ class SoftDeletionModel(db.Model):
     __abstract__ = True
 
     deleted_at = db.Column(db.DateTime(timezone=True))
+
+    def get_new_identifier(self):
```
Comment (iamareebjamal) [inline app/models/base.py:17]: This is not the correct place to put the method. Not all models need this

File: app/models/base.py [hunk 3]
```diff
@@ -9,3 +13,44 @@ class SoftDeletionModel(db.Model):
...
+        :return:
+        """
+        model_name = type(self).__name__
+        with db.session.no_autoflush:
+            if model_name in {'Order', 'EventInvoice'}:
+                identifier = self.random_uuid()
+            elif model_name in {'SocialLink', 'Event'}:
+                identifier = self.random_string_identifier()
+            else:
+                identifier = None
```
Comment (iamareebjamal) [inline app/models/base.py:29]: This is not how to do things, there is no need of if else, just use the correct method in correct model

File: app/models/event_invoice.py [hunk 4]
```diff
@@ -25,7 +12,7 @@ class EventInvoice(SoftDeletionModel):
     __tablename__ = 'event_invoices'
 
     id = db.Column(db.Integer, primary_key=True)
-    identifier = db.Column(db.String, unique=True, default=get_new_identifier)
+    identifier = db.Column(db.String, unique=True)
```
Comment (iamareebjamal) [inline app/models/event_invoice.py:15]: revert

File: app/models/order.py [hunk 5]
```diff
@@ -60,7 +60,7 @@ class Order(SoftDeletionModel):
     __tablename__ = "orders"
 
     id = db.Column(db.Integer, primary_key=True)
-    identifier = db.Column(db.String, unique=True, default=get_new_order_identifier)
+    identifier = db.Column(db.String, unique=True)
```
Comment (iamareebjamal) [inline app/models/order.py:63]: revert

File: app/models/social_link.py [hunk 6]
```diff
@@ -6,28 +6,18 @@
...
-        return get_new_social_link_identifier(length)
-
-
 class SocialLink(SoftDeletionModel):
     __tablename__ = "social_links"
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String, nullable=False)
     link = db.Column(db.String, nullable=False)
-    identifier = db.Column(db.String, default=get_new_social_link_identifier)
+    identifier = db.Column(db.String)
```
Comment (iamareebjamal) [inline app/models/social_link.py:14]: revert

**Comment by jonodrew** [comment]
Thanks for your feedback!

What does 'revert' mean?

**Comment by iamareebjamal** [comment]
https://www.thefreedictionary.com/revert
"To go back to a former condition, practice, subject, or belief"

**Comment by jonodrew** [comment]
Hi!

I appreciate you're really busy, but I don't feel this is an encouraging or pleasant way to approach contributors. I'm going to abandon this. Thanks again for your feedback.

**Comment by iamareebjamal** [comment]
I'm not busy at all. You asked what revert means, I provided the meaning. Best of luck nevertheless, I understand it is not everyone's cup of tea

**Pull request closed without merging**